spotbugs-tests/src/test/java/edu/umd/cs/findbugs/detect/FindIncreasedAccessibilityOfMethodsTest.java

```diff
@@ -0,0 +1,120 @@
+        performAnalysis(
+                "increasedAccessibilityOfMethods/CloneableImplementation.class");
+
+        assertContainsCorrectNumberOfIAOMBugs(0);
+    }
+
+    @Test
+    void findNoIAOMBugInClass_IncreasedAccessibilityOfMethods_InterfaceImplementation() {
+        performAnalysis(
+                "increasedAccessibilityOfMethods/Interface.class",
+                "increasedAccessibilityOfMethods/InterfaceImplementation.class");
+
+        assertContainsCorrectNumberOfIAOMBugs(0);
+    }
+
+    @Test
+    void findNoIAOMBugInClass_IncreasedAccessibilityOfMethods_GenericImplementation() {
+        performAnalysis(
+                "increasedAccessibilityOfMethods/GenericSuperClass.class",
+                "increasedAccessibilityOfMethods/GenericSubClass.class");
+
+        assertContainsCorrectNumberOfIAOMBugs(3);
+
+        assertContainsIAOMBug("GenericSubClass", "protectedMethodToPublicWithGenericParameter");
+        assertContainsIAOMBug("GenericSubClass", "packagePrivateMethodToPublicWithGenericParameter");
+        assertContainsIAOMBug("GenericSubClass", "packagePrivateMethodToProtectedWithGenericParameter");
+    }
+
+    @Test
+    void findNoIAOMBugInClass_IncreasedAccessibilityOfMethods_SubClassWithRepeatedOverride() {
+        performAnalysis(
+                "increasedAccessibilityOfMethods/SuperClassOfSuperClass.class",
+                "increasedAccessibilityOfMethods/SuperClassWithOverride.class",
+                "increasedAccessibilityOfMethods/SubClassWithRepeatedOverride.class");
+
+        assertContainsCorrectNumberOfIAOMBugs(3);
+
+        assertContainsIAOMBug("SuperClassWithOverride", "superProtectedMethodToPublicOverriddenInSuperClass");
+        assertContainsIAOMBug("SuperClassWithOverride", "superPackagePrivateMethodToProtectedOverriddenInSuperClass");
+        assertContainsIAOMBug("SubClassWithRepeatedOverride", "superPackagePrivateMethodToProtectedOverriddenInSuperClass");
+        assertNotContainsIAOMBug("SubClassWithRepeatedOverride", "superProtectedMethodToPublicOverriddenInSuperClass");
+    }
+
+    private void assertContainsCorrectNumberOfIAOMBugs(int numberOfBugs) {
+        BugInstanceMatcher bugTypeMatcher = new BugInstanceMatcherBuilder()
+                .bugType("IAOM_DO_NOT_INCREASE_METHOD_ACCESSIBILITY").build();
+        assertThat(getBugCollection(), containsExactly(numberOfBugs, bugTypeMatcher));
+    }
+
+    private void assertContainsIAOMBug(String className, String methodName) {
+        BugInstanceMatcher bugInstanceMatcher = new BugInstanceMatcherBuilder()
+                .bugType("IAOM_DO_NOT_INCREASE_METHOD_ACCESSIBILITY")
+                .inClass(className)
+                .inMethod(methodName)
+                .build();
+        assertThat(getBugCollection(), hasItem(bugInstanceMatcher));
+    }
+
+    private void assertNotContainsIAOMBug(String className, String methodName) {
```

Comment:
Some assert functions got introduced to `AbstractIntegrationTest` to minimize code repetition. Can you please use `assertNoBugInMethod()` instead of this one and remove this function once it's not used?